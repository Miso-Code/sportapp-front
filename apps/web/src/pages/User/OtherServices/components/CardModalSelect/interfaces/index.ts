import { Product } from '@sportapp/sportapp-repository/src/businessPartner/interfaces'

export interface Props {
	selectedProduct: Product
	quantity: string
	handleQuantityChange: (
		e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
	) => void
	handleClose: () => void
	handleSuccess: () => void
}
