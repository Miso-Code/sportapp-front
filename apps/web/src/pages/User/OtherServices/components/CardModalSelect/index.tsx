import {
	Box,
	Button,
	CardActions,
	CardContent,
	CardMedia,
	TextField,
	Typography
} from '@mui/material'
import MuiMarkdown from 'mui-markdown'
import { Props } from './interfaces'
import { useTranslation } from 'react-i18next'

export default function CardModalSelect({
	selectedProduct,
	quantity,
	handleQuantityChange,
	handleClose
}: Props) {
	const { t } = useTranslation()
	return (
		<CardContent>
			<Typography variant='h6'>{selectedProduct.name}</Typography>
			<CardMedia
				component='img'
				height='240'
				image={selectedProduct.image_url}
				alt={selectedProduct.name}
			/>
			<Box sx={{ mt: 4 }}>
				<MuiMarkdown>{selectedProduct.description}</MuiMarkdown>
			</Box>
			<Box>
				<TextField
					sx={{ mt: 4 }}
					id='quantity'
					name='quantity'
					type='number'
					inputMode='numeric'
					placeholder='0'
					value={quantity}
					fullWidth
					onChange={handleQuantityChange}
					label={t('productService.quantity')}
					variant='outlined'
				/>
			</Box>
			<CardActions
				sx={{
					display: 'flex',
					justifyContent: 'end',
					marginTop: '2rem'
				}}>
				<Button size='small' color='error' onClick={handleClose}>
					{t('productService.close')}
				</Button>
				<Button size='small' onClick={handleClose}>
					{t('productService.buy')}
				</Button>
			</CardActions>
		</CardContent>
	)
}
