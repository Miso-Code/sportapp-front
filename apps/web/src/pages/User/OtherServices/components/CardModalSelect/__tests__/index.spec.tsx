import {
	act,
	fireEvent,
	render,
	RenderResult,
	waitFor
} from '@testing-library/react'
import CardModalSelect from '..'

describe('CardModalSelect', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<CardModalSelect
				selectedProduct={{
					product_id: '6dcba8dc-c9ae-40c7-b42a-21029c7819d8',
					business_partner_id: 'f9368e86-dacd-4904-989a-ba5f23cd25ae',
					category: 'equipement',
					name: 'Awesome Rubber Computer',
					summary:
						'Adflicto sonitus fuga aperio calculus calamitas vicissitudo.',
					url: 'https://apprehensive-poem.net/',
					price: 408.0,
					payment_type: 'unique',
					payment_frequency: 'other',
					image_url: 'https://picsum.photos/seed/wPfJxYXCA/640/480',
					description:
						'Adipiscor demoror velit ventito reiciendis audacia pax abbas aliquam.',
					active: true
				}}
				quantity='1'
				handleQuantityChange={jest.fn()}
				handleClose={jest.fn()}
			/>
		)
	})

	it('should render the component', () => {
		expect(wrapper.container).toMatchSnapshot()
	})

	it('should call handleQuantityChange', async () => {
		const handleQuantityChange = jest.fn()
		wrapper.rerender(
			<CardModalSelect
				selectedProduct={{
					product_id: '6dcba8dc-c9ae-40c7-b42a-21029c7819d8',
					business_partner_id: 'f9368e86-dacd-4904-989a-ba5f23cd25ae',
					category: 'equipement',
					name: 'Awesome Rubber Computer',
					summary:
						'Adflicto sonitus fuga aperio calculus calamitas vicissitudo.',
					url: 'https://apprehensive-poem.net/',
					price: 408.0,
					payment_type: 'unique',
					payment_frequency: 'other',
					image_url: 'https://picsum.photos/seed/wPfJxYXCA/640/480',
					description:
						'Adipiscor demoror velit ventito reiciendis audacia pax abbas aliquam.',
					active: true
				}}
				quantity='1'
				handleQuantityChange={handleQuantityChange}
				handleClose={jest.fn()}
			/>
		)
		const input = wrapper.getByPlaceholderText('0')

		act(() => {
			fireEvent.change(input, { target: { value: '2' } })
		})

		await waitFor(() => {
			expect(handleQuantityChange).toHaveBeenCalled()
		})
	})
})
