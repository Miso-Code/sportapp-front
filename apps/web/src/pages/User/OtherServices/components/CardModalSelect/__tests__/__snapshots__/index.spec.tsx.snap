// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardModalSelect should render the component 1`] = `
<div>
  <div
    class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Awesome Rubber Computer
    </h6>
    <img
      alt="Awesome Rubber Computer"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="240"
      src="https://picsum.photos/seed/wPfJxYXCA/640/480"
    />
    <p
      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      Adflicto sonitus fuga aperio calculus calamitas vicissitudo.
    </p>
    <div
      class="MuiBox-root css-h5fkc8"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 css-16gwivt-MuiTypography-root"
      >
        Adflicto sonitus fuga aperio calculus calamitas vicissitudo.
      </span>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-12r883s-MuiFormControl-root-MuiTextField-root"
        inputmode="numeric"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="quantity"
          id="quantity-label"
        >
          productService.quantity
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="quantity"
            name="quantity"
            placeholder="0"
            type="number"
            value="1"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                productService.quantity
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1sci87t-MuiCardActions-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError css-6ekzlo-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        productService.close
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        productService.buy
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
